app/json_schemas/customer_schema.rb

```diff
@@ -61,4 +61,18 @@ def self.writable_attributes
   def self.relationships
     [:enterprise]
   end
+
+  # Optional attributes included with "with"
+  def self.balance
+    {
+      balance: {
+        type: :object,
+        required: [:value, :time],
+        properties: {
+          value: { type: :number, format: :double },
+          time: { type: :string, format: "date-time" }
```

Comment:
I'm not sure about including the time within a new object. That's a cool idea and knowing the time could be really helpful but maybe it would be even more helpful to have a new field on the customer itself, for example `viewed_at`. Other properties can change over time as well. And if you see the whole thing like a "report" then it makes sense to have that attribute on the top level.

On the other hand, the querying program can just note the timestamp of retrieval as well.

What do you think, @lin-d-hop?

Reply:
I guess we'd find it helpful for debugging to include the time, but I agree @mkllnk that it can be on the top level. I don't have a strong opinion here.

Reply:
Responded in comment below.

Reply:
Time removed